tree/tree/src/TTree.cxx

```diff
@@ -5279,6 +5281,9 @@ TBranch *R__GetBranch(const TObjArray &branches, const char *name)
       if (!strcmp(b->GetName(), name)) {
          return b;
       }
+      const std::string prefix{b->GetName()};
+      if (prefix.compare(0, prefix.size(), name, prefix.size()) != 0)
+         continue;
```

Comment:
This is seem odd.  Which cases does it support?  Does it work correctly if the top level branch is elided from the name search or the branch name (i.e. 'no trailing dot' case)?

Reply:
It was a smaller optimization I was playing with before the unordered_map. It probably won't be necessary so I think I will remove it.

Reply:
Should we update the summary (and git log) with the latest results? :)

Reply:
The git log was already updated.  Thanks.  I updated the summary.